tests/unit/s2n_client_supported_versions_extension_test.c

```diff
@@ -396,6 +392,62 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_connection_free(conn));
     }
 
+    /**
+     *= https://tools.ietf.org/rfc/rfc8446#appendix-D.2
+     *= type=test
+     *# A TLS server can also receive a ClientHello indicating a version number smaller than its highest supported
+     *# version. If the "supported_versions" extension is present, the server MUST negotiate using that extension as
+     *#  described in Section 4.2.1.
+     */
+    if (s2n_is_tls13_fully_supported()) {
+        {
+            DEFER_CLEANUP(struct s2n_cert_chain_and_key * chain_and_key,
+                    s2n_cert_chain_and_key_ptr_free);
+            EXPECT_SUCCESS(s2n_test_cert_chain_and_key_new(&chain_and_key,
+                    S2N_DEFAULT_ECDSA_TEST_CERT_CHAIN,
+                    S2N_DEFAULT_ECDSA_TEST_PRIVATE_KEY));
+
+            DEFER_CLEANUP(struct s2n_config *configNew = s2n_config_new(),
```

Comment:
If you can't call this "config" I would call it "config_with_cert" because that is what makes it different from the previous config. Also we use snake case in this library.